packages/corvid-cli/src/utils/electron.js

```diff
@@ -29,68 +29,54 @@ const beforeCloseDialogParams = {
 const runningProcesses = [];
 
 function launch(file, options = {}, callbacks = {}, args = []) {
-  options.env = {
-    ...process.env,
-    ...options.env,
-    FORCE_COLOR: "yes"
-  };
-
-  const cp = childProcess.spawn(
+  const childElectronProcess = childProcess.spawn(
     electron,
     [path.resolve(path.join(file)), ...args],
     {
-      ...options
+      ...options,
+      env: {
+        ...process.env,
+        ...options.env,
+        FORCE_COLOR: "yes"
+      },
+      stdio: ["inherit", "inherit", "inherit", "ipc"]
     }
   );
-  runningProcesses.push(cp);
+  runningProcesses.push(childElectronProcess);
 
-  function isJunk(data) {
-    const text = data.toString();
-    return text.includes("Could not instantiate: ProductRegistryImpl.Registry");
-  }
+  const removeProcess = () =>
+    runningProcesses.splice(runningProcesses.indexOf(childElectronProcess), 1);
 
   return new Promise((resolve, reject) => {
-    const messages = [];
-
-    if (options.detached) {
-      cp.unref();
-      resolve();
-    } else {
-      if (options.stdio !== "ignore") {
-        cp.stdout.on("data", function(data) {
-          if (isJunk(data)) {
-            return;
-          }
-          try {
-            const msg = JSON.parse(data);
-            if (typeof msg === "object") {
-              messages.push(msg);
-              if (msg.event && typeof callbacks[msg.event] === "function") {
-                callbacks[msg.event](msg.payload);
-              } else if (msg.event === "error") {
-                reject(new Error(msg.payload));
-              }
-            }
-          } catch (_) {
-            return;
+    childElectronProcess.on("message", message => {
+      if (typeof message === "object") {
```

Comment:
Personally, I would prefer lodash's `isFunction`/`isObject` to the use of `typeof`, just so it's easier to read